pkg/services/gitops/uninstall_test.go

```diff
@@ -22,38 +73,91 @@ var _ = Describe("Uninstall", func() {
 				return kube.WeGOInstalled
 			},
 		}
-		gitopsSrv = gitops.New(&loggerfakes.FakeLogger{}, fluxClient, kubeClient)
+		logger = &loggerfakes.FakeLogger{}
+		gitopsSrv = gitops.New(logger, fluxClient, kubeClient)
 
 		uninstallParams = gitops.UinstallParams{
 			Namespace: "wego-system",
 			DryRun:    false,
 		}
 	})
 
-	It("checks if wego is installed before proceeding", func() {
+	It("logs warning information if wego is not installed before proceeding", func() {
 		err := gitopsSrv.Uninstall(uninstallParams)
 		Expect(err).ShouldNot(HaveOccurred())
 
 		Expect(kubeClient.GetClusterStatusCallCount()).To(Equal(1))
 		Expect(fluxClient.UninstallCallCount()).To(Equal(1))
 
+		loggedMsg := ""
+		logger.PrintlnStub = func(str string, args ...interface{}) {
+			loggedMsg = str
+		}
+
+		kubeClient.GetClusterStatusStub = func(ctx context.Context) kube.ClusterStatus {
+			return kube.FluxInstalled
+		}
+
+		Expect(gitopsSrv.Uninstall(uninstallParams)).Should(Succeed())
+		Expect(loggedMsg).To(Equal("wego is not fully installed... removing any partial installation\n"))
+
 		kubeClient.GetClusterStatusStub = func(ctx context.Context) kube.ClusterStatus {
 			return kube.Unmodified
 		}
+		loggedMsg = ""
 
-		err = gitopsSrv.Uninstall(uninstallParams)
-		Expect(err).Should(MatchError("wego is not installed... exiting"))
+		Expect(gitopsSrv.Uninstall(uninstallParams)).Should(Succeed())
+		Expect(loggedMsg).To(Equal("wego is not fully installed... removing any partial installation\n"))
 	})
 
-	It("calls flux uninstall", func() {
-		err := gitopsSrv.Uninstall(uninstallParams)
-		Expect(err).ShouldNot(HaveOccurred())
+	It("Does not log warning information if wego is installed", func() {
+		loggedMsg := ""
+		logger.PrintlnStub = func(str string, args ...interface{}) {
+			loggedMsg = str
+		}
 
-		Expect(fluxClient.UninstallCallCount()).To(Equal(1))
+		Expect(gitopsSrv.Uninstall(uninstallParams)).Should(Succeed())
+		Expect(loggedMsg).To(Equal(""))
+	})
+
+	It("Generates an error if flux uninstall fails with wego installed", func() {
+		checkFluxUninstallFailure()
+	})
+
+	It("Generates an error if flux uninstall fails with only flux installed", func() {
+		kubeClient.GetClusterStatusStub = func(ctx context.Context) kube.ClusterStatus {
+			return kube.FluxInstalled
+		}
+
+		checkFluxUninstallFailure()
+	})
+
+	It("Generates an error if flux uninstall fails with partial or no flux installed", func() {
+		kubeClient.GetClusterStatusStub = func(ctx context.Context) kube.ClusterStatus {
+			return kube.Unmodified
+		}
+
+		checkFluxUninstallFailure()
+	})
+
+	It("Generates an error if CRD uninstall fails with wego installed", func() {
+		checkAppCRDUninstallFailure()
```

Comment:
I'm having a tough time reasoning about these tests. I would guess this is trying to stay DRY, but a test case with no assertions and no arguments into a helper function makes me scratch my head a little bit.

Reply:
Understood... the only difference between the test cases, though, is the cluster status and this case has it set by default. I should probably set it explicitly here to make that clear.